src/modules/ie-objects/helpers/limit-access-to-object-details.spec.ts

```diff
@@ -10,35 +10,261 @@ import {
+	it('Test case 1 - user ziet uitgebreide metadataset op de detailpagina', () => {
+		const limitedAccessIeObject = limitAccessToObjectDetails(
+			// DPG Media (sector = RURAL) - INTRA LICENSES + VIAA-PUBLIEK_METADATA_ALL
+			{
+				...mockIeObject,
+				sector: IeObjectSector.RURAL,
+				licenses: [IeObjectLicense.PUBLIEK_METADATA_ALL, ...IE_OBJECT_INTRA_CP_LICENSES],
+			},
+			// Basic user - isKeyUser - Public Sector
+			{
+				...mockUserInfo,
+				groupId: Group.VISITOR,
+				isKeyUser: true,
+				sector: IeObjectSector.PUBLIC,
+				accessibleVisitorSpaceIds: [],
+			}
+		);
+
+		expect(limitedAccessIeObject).toEqual({
+			...mockIeObjectWithMetadataSetALL,
+			licenses: [IeObjectLicense.PUBLIEK_METADATA_ALL, ...IE_OBJECT_INTRA_CP_LICENSES],
+			accessThrough: [IeObjectAccessThrough.SECTOR, IeObjectAccessThrough.PUBLIC_INFO],
+		});
+	});
+
+	it('Test case 2 - user ziet uitgebreide metadataset en essence op de detailpagina', () => {
+		const limitedAccessIeObject = limitAccessToObjectDetails(
+			// ADVN (sector: culture) - INTRA LICENSES + VIAA-PUBLIEK_METADATA_LTD
+			{
+				...mockIeObject,
+				sector: IeObjectSector.CULTURE,
+				licenses: [IeObjectLicense.PUBLIEK_METADATA_LTD, ...IE_OBJECT_INTRA_CP_LICENSES],
+			},
+			// CP admin works for ADVN - hasVisitorSpace - Culture sector
+			{
+				...mockUserInfo,
+				groupId: Group.CP_ADMIN,
+				isKeyUser: true,
+				sector: IeObjectSector.CULTURE,
+			}
+		);
+
+		expect(limitedAccessIeObject).toEqual({
+			...mockIeObjectWithMetadataSetALLWithEssence,
+			licenses: [IeObjectLicense.PUBLIEK_METADATA_LTD, ...IE_OBJECT_INTRA_CP_LICENSES],
+			accessThrough: [
+				IeObjectAccessThrough.SECTOR,
+				IeObjectAccessThrough.VISITOR_SPACE_FULL,
+				IeObjectAccessThrough.PUBLIC_INFO,
+			],
+		});
+	});
+
+	it('Test case 3 - user ziet uitgebreide metadataset en essence op de detailpagina', () => {
+		const limitedAccessIeObject = limitAccessToObjectDetails(
+			// VRT - VIAA-PUBLIEK_METADATA_LTD + VIAA-INTRA_CP-METADATA-ALL + BEZOEKERTOOL-CONTENT
+			{
+				...mockIeObject,
+				sector: IeObjectSector.CULTURE,
+				licenses: [
+					IeObjectLicense.PUBLIEK_METADATA_LTD,
+					IeObjectLicense.INTRA_CP_METADATA_ALL,
+					IeObjectLicense.BEZOEKERTOOL_CONTENT,
+				],
+			},
+			// Basis gebruiker - isKeyUser - Culture sector - temporary access visitor space VRT
+			{
+				...mockUserInfo,
+				groupId: Group.VISITOR,
+				isKeyUser: true,
+				sector: IeObjectSector.CULTURE,
+			}
+		);
+
+		expect(limitedAccessIeObject).toEqual({
+			...mockIeObjectWithMetadataSetALLWithEssence,
+			licenses: [
+				IeObjectLicense.PUBLIEK_METADATA_LTD,
+				IeObjectLicense.INTRA_CP_METADATA_ALL,
+				IeObjectLicense.BEZOEKERTOOL_CONTENT,
+			],
+			accessThrough: [
+				IeObjectAccessThrough.SECTOR,
+				IeObjectAccessThrough.VISITOR_SPACE_FULL,
+				IeObjectAccessThrough.PUBLIC_INFO,
+			],
+		});
+	});
+
+	it('Test case 4 - user ziet uitgebreide metadataset op de detailpagina', () => {
```

Comment:
it might also be helpful to test this exact same test 4b
but with a different accessibleObjectIdsThroughFolders array to see if then the BEZOEKERTOOL_METADATA_ALL is not present